File: core/src/main/java/com/mcreater/amclcore/concurrent/task/model/BooleanTask.java
package com.mcreater.amclcore.concurrent.task.model;

import com.mcreater.amclcore.concurrent.task.AbstractTask;
import com.mcreater.amclcore.i18n.Text;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;

import static com.mcreater.amclcore.i18n.I18NManager.translatable;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class BooleanTask extends AbstractTask<Boolean> {
    private boolean result;
    private Text text;

    protected Boolean call() {
        return result;
    }

    protected Text getTaskName() {
        return text;
    }

    public static BooleanTask of(boolean b) {
        return new BooleanTask(b, translatable("core.concurrent.task.model.empty"));
    }

    public static BooleanTask of(boolean b, Text text) {
        return new BooleanTask(b, text);
    }
}

File: core/src/main/java/com/mcreater/amclcore/concurrent/task/model/BooleanTask.kt
package com.mcreater.amclcore.concurrent.task.model

import com.mcreater.amclcore.concurrent.task.AbstractTask
import com.mcreater.amclcore.i18n.I18NManager.Companion.translatable
import com.mcreater.amclcore.i18n.Text


data class BooleanTask(
    val result: Boolean = false,
    val text: Text? = null
) : AbstractTask<Boolean?>() {

    override fun call(): Boolean {
        return result
    }

    override fun getTaskName(): Text {
        return text!!
    }

    companion object {
        @JvmStatic
        fun of(b: Boolean): BooleanTask {
            return BooleanTask(b, translatable("core.concurrent.task.model.empty"))
        }

        @JvmStatic
        fun of(b: Boolean, text: Text?): BooleanTask {
            return BooleanTask(b, text)
        }
    }
}

